### PR TITLE
Abstract-out-template

### LINKED ISSUE #17: Refactor the create_badge function to source the template from a tex file instead of being hard coded to the function
Currently `create_badge` is an 'omni-function' in that is 'does it all'. There are various interfaces that can be broken out, and this could start with a built in `tex` file to the package which contains the default badge template. Being able to break this out might make it easier for the package to cover multiple different styles of badge, and to potentially maintain and modify the behaviour fo the template.

This work will begin with creating a standalone tex file in a suitable location in the package and then making sure that is sourced correctly into the function.

### PR DESCRIPTION
**Pull request by DaveParr**
A simple method to abstract out the template tex string from the main body function.

In this method the package has a semi-compliant tex file, with syntax highlighting that can be opened and inspected independantly.

By doing this we open up the possibility of supplying an interface for users to provide custom tex templates (within reason + compliant with the glue interpolation mechanisims) to enable more customisation, such as portrait style badges etc.

This 'works' but the `readLines` + `collapse` feels minorly janky. Open to suggestions for improvement in reading in `.tex` but personally couldn't find a better solutions without increasing the dependencies of the package with other packages of semi-debatable quality. Would have thought knitr/tinytex might have had an internal tex reading method I could yank but couldn't turn anything up.

### CODE REVIEW

File: tex/default_template.tex [hunk 1]
```diff
@@ -0,0 +1,27 @@
+\documentclass[a4paper,10pt]{{letter}}
```
Comment (dokato) [inline tex/default_template.tex]: this should be single curly bracket, see: https://gist.github.com/dokato/fcf2445c16135802c4404f8a90bbd22a

File: tex/default_template.tex [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+\usepackage[utf8]{{inputenc}}
```
Comment (dokato) [inline tex/default_template.tex]: single curly brackets

File: tex/default_template.tex [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+\usepackage[freepin,{edge_type}]{{ticket}}
```
Comment (dokato) [inline tex/default_template.tex]: single curly bracket

File: tex/default_template.tex [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+\usepackage{{graphicx}}
```
Comment (dokato) [inline tex/default_template.tex]: single curly brackets

File: tex/default_template.tex [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+\unitlength=1mm
+\ticketSize{{{badge_width}}}{{{badge_height}}}
+\ticketNumbers{{{cards_per_page_x}}}{{{cards_per_page_y}}}
+
+\renewcommand{{\ticketdefault}}{{
```
Comment (dokato) [inline tex/default_template.tex]: single curly brackets

File: tex/default_template.tex [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+}}
+
+% now what do you like to put in your ticket
+\newcommand{{\confpin}}[3]{{\ticket{{%
+\put({main_text_x},{main_text_y+20}){{\makebox[0mm]{{\bfseries\huge #1}}}}
+\put({main_text_x},{main_text_y+10}){{\makebox[0mm]{{\bfseries\huge #2}}}}
+\put({main_text_x},{main_text_y}){{\makebox[0mm]{{\large \textit{{#3}}}}}}
+}}}}
+
+\begin{{document}}
```
Comment (dokato) [inline tex/default_template.tex]: single curly brackets

File: tex/default_template.tex [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+\sffamily
+{cards}
+\end{{document}}
```
Comment (dokato) [inline tex/default_template.tex]: single curly brackets

**Review by dokato — CHANGES_REQUESTED** [review]

**Comment by dokato** [comment]
This looks actually exactly like something I had in mind. I didn't know it would be so easy with readLines. In general, I really like how it looks right already (apart from minor edits in the `.tex` template. The only confusing bit is ambiguity between latex curly brackets and glue curly brackets, but I guess we can live with that!

**Comment by DaveParr** [comment]
So do you want to merge this in, or shall I move some of the file location/ documentation?

**Comment by dokato** [comment]
Yes, we could merge that in. Just please fix the conflicts and address the issues above before, thanks :)

File: tex/default_template.tex (hunk 5 above)
Comment (DaveParr) [reply tex/default_template.tex]: Here is it by design there are triple `{}` for the variables? Should they be double? Assuming you mean the `\renewcommand{{\ticketdefault}}` should be `\renewcommand{\ticketdefault}`

Comment (dokato) [reply tex/default_template.tex]: Yeap, single, look for comparison: https://gist.github.com/dokato/fcf2445c16135802c4404f8a90bbd22a

**Review by dokato — APPROVED** [review]
LGTM

**Pull request merged**